gremlin-dotnet/glv/generate.groovy

```diff
@@ -48,7 +48,7 @@ def toCSharpTypeMap = ["Long": "long",
                        "TraversalMetrics": "E2",
                        "Traversal": "ITraversal",
                        "Traversal[]": "ITraversal[]",
-                       "Predicate": "TraversalPredicate",
+                       "Predicate": "object",
```

Comment:
I think we should use an interface `ILambda` similar to java's counterpart: https://github.com/apache/tinkerpop/blob/3.2.7/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/util/function/Lambda.java#L33

`Lambda.Groovy()` and `Lambda.Python()` could return `ILambda` instances. That way we can have a compile type check for methods like `Filter()`, `Until()`, etc... to accept only `ILambda` instances.

Reply:
I don't know if you saw what I wrote in the initial post of this PR, but the problem is that those steps expect a `Function` in Java which is also implemented by enums like `T`. So, when we restrict them to something like `ILambda` in .NET then they can't take enum values anymore as enums can't implement interfaces in .NET.

Reply:
You're right, we should solve this problem, but I think that the changes for that are big enough to validate its own issue and it's also somewhat unrelated to this PR. So I created a new issue for that: [TINKERPOP-1901](https://issues.apache.org/jira/browse/TINKERPOP-1901). I suggest that we resolve that one first and then I simply rebase this PR afterwards and let the `Lambda` class implement whichever interface we introduce there for those cases.

Reply:
Makes sense 👍